### PR TITLE
fix: slug generation uses actor locale instead of forum default

### PR DESCRIPTION
**Pull request by luceos**
Fixes the issue with generating slugs based on actor locale not forum default locale.

### CODE REVIEW

File: framework/core/tests/integration/api/discussions/CreateTest.php [hunk 1]
```diff
@@ -138,7 +138,40 @@ public function can_create_discussion()
      */
     public function can_create_discussion_with_current_lang_slug_transliteration()
     {
-        $this->app()->getContainer()->make('flarum.locales')->setLocale('zh');
+        // Forum default is traditional Chinese.
+        $this->app()->getContainer()->make('flarum.settings')->set('default_locale', 'zh');
```
Comment (SychO9) [inline framework/core/tests/integration/api/discussions/CreateTest.php]: let's use the existing method to change settings in tests:
```suggestion
        $this->setting('default_locale', 'zh');
```

File: framework/core/tests/integration/api/discussions/CreateTest.php [hunk 2]
```diff
@@ -138,7 +138,40 @@ public function can_create_discussion()
...
+        $this->assertEquals('wo-shi-yi-ge-tu-dou', $discussion->slug);
+    }
+
+    /**
+     * @test
+     */
+    public function can_create_discussion_with_forum_locale_transliteration()
+    {
+        // Forum default is traditional Chinese.
+        $this->app()->getContainer()->make('flarum.settings')->set('default_locale', 'zh');
```
Comment (SychO9) [inline framework/core/tests/integration/api/discussions/CreateTest.php]:
```suggestion
        $this->setting('default_locale', 'zh');
```

**Review by SychO9 — COMMENTED** [review]
Looks good! just a minor suggestion

File: framework/core/tests/integration/api/discussions/CreateTest.php (hunk 1 above)
Comment (luceos) [reply framework/core/tests/integration/api/discussions/CreateTest.php]: Oh wow I copied that from another test 🙈

**Review by SychO9 — APPROVED** [review]

**Review by tankerkiller125 — APPROVED** [review]

**Pull request merged**